src/site/_includes/partials/analytics.njk

```diff
@@ -1,14 +1,11 @@
 {% set analyticsScript %}
-window.ga =
-  window.ga ||
-  function () {
-    (ga.q = ga.q || []).push(arguments);
-  };
-ga.l = +new Date();
-ga('create', '{{ site.analytics.ids.prod }}');
-ga('set', 'transport', 'beacon');
-ga('set', 'anonymizeIp', true);
-ga('set', 'page', window.location.pathname);
-ga('set', '{{ site.analytics.dimensions.TRACKING_VERSION }}', '{{ site.analytics.version}}');
+window.dataLayer = window.dataLayer || [];
+function gtag(){dataLayer.push(arguments);}
+gtag('js', new Date());
+loadScript('https://www.googletagmanager.com/gtag/js?id={{ site.analytics.ids.ga4 }}', null);
+{% if (site.env != 'prod') %}
+  {# In development, log debug events to the console instead of sending them to GA #}
+  window.gtag = console.debug;
+{% endif %}
```

Comment:
Should we only load the script if in prod mode?

And is it worth noting that googletagmanager call will actually load both scripts (based on the config) and not just the ga4 scripts? that confused me initially.

And is there any risk that the googletagmanager script will override window.gtag after it loads asynchronously? It doesn't seem to but just wondering? Maybe another reason to only load that if needed?


```suggestion
{% if (site.env === 'prod') %}
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  {# Note we only need one call to GTM and it will load all the analytics from config object #}
  loadScript('https://www.googletagmanager.com/gtag/js?id={{ site.analytics.ids.ga4 }}', null);
{% else %}
  {# In development, log debug events to the console instead of sending them to GA #}
  window.gtag = console.debug;
{% endif %}
```

Reply:
I'll move the loading of the script into the prod conditional in my next commit. We do, however, need to ensure the `gtag()` function is always defined since other scripts call it.